test/e2e/httpproxy/request_redirect_test.go

```diff
@@ -52,40 +52,55 @@ func testRequestRedirectRule(namespace string) {
 	})
 }
 
+func testRequestRedirectRuleInvalid(namespace string) {
+	Specify("invalid policy specified on route rule", func() {
+
+		f.Fixtures.Echo.Deploy(namespace, "echo")
+		proxy := getHTTPProxyInvalid(namespace)
+
+		f.CreateHTTPProxyAndWaitFor(proxy, e2e.HTTPProxyInvalid)
+	})
+}
+
 func doTest(namespace string, proxy *contour_api_v1.HTTPProxy, t GinkgoTInterface) {
```

Comment:
Since these helpers are scoped to the entire `test/e2e/httpproxy` package some more specific naming would help:

```suggestion
func doRedirectTest(namespace string, proxy *contour_api_v1.HTTPProxy, t GinkgoTInterface) {
```